### PR TITLE
update messages for old modules

### PR DESCRIPTION
**Pull request by smoors**
tested in login3, works as advertized.

### CODE REVIEW

File: SitePackage.lua [hunk 1]
```diff
@@ -44,15 +44,25 @@ end
 local function old_module_check(modT)
     -- Check if a module is 'old'
     -- modT should have a entry 'fn' with the module path
-    -- returns true/false
+    -- returns 4 values: true/false, cutoff1, true/false, cutoff2
```
Comment (wpoely86) [inline SitePackage.lua]: looking at this, I would simply bit to a function that returns the 'age' of the module in terms of halve years?

Convert `a` to Jan 1 and `b` to July 1 and substract with the current timestamp divide by 6 months?

The the logic is simply: if age > 7 else if age > 5 ?

Comment (smoors) [reply SitePackage.lua]: you still have to calculate the cutoff toolchain generation, so this does not really help.
I've updated it to use timestamps, as that makes it more general indeed.

Comment (wpoely86) [reply SitePackage.lua]: Hmm, I still find it very complex code for something that (at least in my mind) should be simple. Let me think about it for a moment.

Is it worthwhile to keep this cutoff thing in the message? We want them to preferably use the newest toolchain?

Comment (smoors) [reply SitePackage.lua]: to me it's not complex at all, just a lot of boilerplate.

but yeah it makes sense to remove the cutoff toolchain generation from the message.

File: SitePackage.lua [hunk 2]
```diff
@@ -41,18 +41,23 @@ local function logmsg(logTbl)
...
-local function old_module_check(modT)
-    -- Check if a module is 'old'
+local function module_age(modT)
+    -- Calculate the age of a module, relative to the current toolchain generation
     -- modT should have a entry 'fn' with the module path
-    -- returns true/false
+    -- returns age (in months) between module toolchain generation and current toolchain generation
 
     local tcver = modT.fn:match("^/apps/brussel/.*/modules/(20[0-9][0-9][ab])/all/")
     if tcver == nil then return end
```
Comment (wpoely86) [inline SitePackage.lua]: now I see this line I'm wondering about it :p 
Should we return 0 here?

File: SitePackage.lua [hunk 3]
```diff
@@ -41,18 +41,23 @@ local function logmsg(logTbl)
...
 
-    -- always the the a version of two years ago
-    local cutoff = string.format("%da", os.date("%Y") - 2)
+    local suffixmonth = {a=1, b=7}
 
-    return parseVersion(tcver) < parseVersion(cutoff)
+    local tcyear = string.sub(tcver, 1, 4)
+    local tcsuffix = string.sub(tcver, 5, 5)
```
Comment (wpoely86) [inline SitePackage.lua]: shouldn't we do this in the regex for `tcver`? Now we have a capture group for the entire thing, split it out there immediately?

File: SitePackage.lua (hunk 2 above)
Comment (smoors) [reply SitePackage.lua]: I think better not.
if it's nil, something is wrong and it should fail.
if we return 0 we will never know something wrong.

Comment (wpoely86) [reply SitePackage.lua]: I'm more thinking of the case EESSI or a local module path in your account.  This thing should just be ignored completely in those cases. 

It used to return either true/false and `nil` probably counted as false.

Comment (smoors) [reply SitePackage.lua]: ah, indeed.

**Review by wpoely86 — APPROVED** [review]

**Pull request merged**